### PR TITLE
feat: Conditional Input History - Forms-form

### PR DESCRIPTION
**Pull request by thiessenp-cds**
# Conditional Input History - Forms-form

Adds conditional input history to the forms-forms. 
This adda the logic from group history and show-hide to only include the data that follows a user's branching path.

**_High Risk PR as this touches forms-forms (how a user submits data)_**

## Testing

The main things to test are that nothing obvious is broken around form submissions. Also test that the correct data is submitted. See the below video for an example.

Also test that old forms work as they previously did. So a review page should not be displayed when submitting and submissions continue working as previously.

https://github.com/user-attachments/assets/9a994947-7d0f-4f20-8781-0e836c93bc0d

### CODE REVIEW

File: components/clientComponents/forms/Form/Form.tsx [hunk 1]
```diff
@@ -324,10 +332,35 @@ export const Form = withFormik<FormProps, Responses>({
...
+        values.matchedIds as string[]
+      );
+      return filterValuesByShownElements(inputHistoryValues, shownElements);
+    };
+
     // Needed so the Loader is displayed
     formikBag.setStatus("submitting");
     try {
-      const result = await submitForm(values, formikBag.props.language, formikBag.props.formRecord);
+      const isGroupsCheck = formikBag.props.allowGrouping;
```
Comment (timarney) [inline components/clientComponents/forms/Form/Form.tsx:351]: Need to ensure this matches 

```
const isGroupsCheck = groupsCheck(props.allowGrouping);
```

Which check for allow `allowGrouping` flag but also if the form has groups

File: components/clientComponents/forms/Form/Form.tsx [hunk 2]
```diff
@@ -324,10 +332,35 @@ export const Form = withFormik<FormProps, Responses>({
...
+      const isShowHideRules = (values.matchedIds as string[])?.length > 0;
```
Comment (timarney) [inline components/clientComponents/forms/Form/Form.tsx:352]: Might want to call this `hasShowHideRules` ?

**Comment by thiessenp-cds** [comment]
**PS: For testing, you can test on Staging (the changes have been merged)**

**Comment by anikbrazeau** [comment]
Tested this and it remembered my inputs in the draft preview/test, but did not submit the old data when I switched trigger answer to go down a different path. 

I did notice one issue: in the response files (HTML, CSV and JSON), it does appear that answers are not in the same order as they are filled out by the user once submitted? @thiessenp-cds 

Screenshot 2024-07-30 at 8 16 46 AM

**Review by timarney — DISMISSED** [review]

**Review by wmoussa-gc — APPROVED** [review]
I cannot find any obvious flaws; it seems good.

**Review by craigzour — APPROVED** [review]
Same thing for me. I could not find any blocker!

**Review by timarney — APPROVED** [review]

**Review by patheard — APPROVED** [review]

**Pull request merged**